Modules/_functoolsmodule.c

```diff
@@ -41,13 +43,88 @@ get_functools_state(PyObject *module)
 
 /* partial object **********************************************************/
 
+
+// The 'Placeholder' singleton indicates which formal positional
+// parameters are to be bound first when using a 'partial' object.
+
+typedef struct {
+    PyObject_HEAD
+} placeholderobject;
+
+static inline _functools_state *
+get_functools_state_by_type(PyTypeObject *type);
+
+PyDoc_STRVAR(placeholder_doc,
+"The type of the Placeholder singleton.\n\n"
+"Used as a placeholder for partial arguments.");
+
+static PyObject *
+placeholder_repr(PyObject *op)
+{
+    return PyUnicode_FromString("Placeholder");
```

Comment:
Maybe use `functools.Placeholder` instead actually? The singleton is really tight to the functools module and is not just a universal  one. So I think it's closer to what `typing.NoDefault` is.

In particular, you could also decide whether the singleton should be implemented (and exported) using the same logic as for `typing.NoDefault` or if it should be done as for `None` and other global singletons.

(Sorry, I only remembered about that singleton now and not before)